bitsail-connectors/connector-kudu/src/main/java/com/bytedance/bitsail/connector/kudu/source/split/strategy/PartitionDivideSplitConstructor.java

```diff
@@ -0,0 +1,176 @@
+  private static final Logger LOG = LoggerFactory.getLogger(PartitionDivideSplitConstructor.class);
+
+  private SplitConfiguration splitConf = null;
+  private boolean available = false;
+  private transient List<Partition> partitions = null;
+
+  public PartitionDivideSplitConstructor(BitSailConfiguration jobConf, KuduClient client) throws Exception {
+    super(jobConf, client);
+
+    if (!jobConf.fieldExists(KuduReaderOptions.SPLIT_CONFIGURATION)) {
+      LOG.warn("{} cannot work due to lack of split configuration.", this.getClass().getSimpleName());
+      return;
+    }
+    String splitConfStr = jobConf.get(KuduReaderOptions.SPLIT_CONFIGURATION);
+    this.splitConf = new ObjectMapper().readValue(splitConfStr, SplitConfiguration.class);
+    KuduTable kuduTable = client.openTable(this.tableName);
+    this.available = splitConf.isValid(kuduTable);
+    if (!available) {
+      LOG.warn("{} cannot work because split configuration is invalid.", this.getClass().getSimpleName());
+      return;
+    }
+    this.available = fillSplitConf(jobConf, client);
+    if (!available) {
+      return;
+    }
+    LOG.info("Successfully created ,final split configuration is: {}", splitConf);
+  }
+
+  @Override
+  public boolean isAvailable() {
+    return available;
+  }
+
+  @Override
+  @SuppressWarnings("checkstyle:MagicNumber")
+  protected boolean fillSplitConf(BitSailConfiguration jobConf, KuduClient client) throws Exception {
+    if (splitConf.getSplitNum() == null || splitConf.getSplitNum() <= 0) {
+      splitConf.setSplitNum(jobConf.getUnNecessaryOption(KuduReaderOptions.READER_PARALLELISM_NUM, 1));
+    }
+    if (splitConf.timeOutMills == null || splitConf.timeOutMills <= 0) {
+      splitConf.setTimeOutMills(3000L);
+    }
+    KuduTable kuduTable = client.openTable(this.tableName);
+    this.partitions = kuduTable.getRangePartitions(splitConf.getTimeOutMills());
+
+    if (this.partitions.size() < splitConf.getSplitNum()) {
+      this.splitConf.setSplitNum(this.partitions.size());
+      LOG.info("Resize split num to {}.", splitConf.getSplitNum());
+    }
+    return true;
+  }
+
+  @Override
+  public List <KuduSourceSplit> construct(KuduClient kuduClient) throws Exception {
+    List<KuduSourceSplit> splits = new ArrayList <>();
+
+    KuduTable kuduTable = kuduClient.openTable(this.tableName);
+    PartitionSchema partitionSchema = kuduTable.getPartitionSchema();
+    List<Integer> partitionColumnIds = partitionSchema.getRangeSchema().getColumnIds();
+
+    for (int i = 0; i < this.partitions.size(); i++) {
+      KuduSourceSplit split = new KuduSourceSplit(i);
+      for (int j = 0; j < partitionColumnIds.size(); j++) {
+        split.addPredicate(KuduPredicate.newComparisonPredicate(
+            schema.getColumnByIndex(schema.getColumnIndex(partitionColumnIds.get(j))),
+            KuduPredicate.ComparisonOp.GREATER_EQUAL,
+            partitions.get(i).getPartitionKeyStart()[j]));
```

Comment:
Actually, the problem is Partition::getPartitionKeyStart method returns a byte array. So partitions.get(i).getPartitionKeyStart()[j] returns a single byte, which is against expectation.

Reply:
@BlockLiu There must be some wrong with my ideas, I will think about it again, thank you for your suggestions